File: app/src/main/java/com/example/todoapp/adapter/NotesAdapter.java
package com.example.todoapp.adapter;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.todoapp.R;
import com.example.todoapp.clicklisteners.ItemClickLstener;
import com.example.todoapp.model.Notes;

import org.w3c.dom.Text;

import java.util.List;

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {
    List<Notes> listnotes;
    ItemClickLstener itemClickLstener;
    public NotesAdapter(){

   }

   public void setListnotes(List<Notes> list){
       this.listnotes=list;
   }

    @NonNull
    @Override
    public NotesAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view= LayoutInflater.from(parent.getContext()).inflate(R.layout.notes_adapter_layout,parent,false);
        return  new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NotesAdapter.ViewHolder holder, int position) {
        //we set data from list to layout here
        Notes notes= listnotes.get(position);
        String ttle= notes.getTitle();
        String desc=notes.getDesc();
        holder.textView_ttle.setText(ttle);
        Log.d("TAGVAL","ttle"+ttle);
        Log.d("TAGVAL","ttle"+desc);
        holder.textView_desc.setText(desc);

    }

    @Override
    public int getItemCount() {
        if (listnotes!=null)
            return listnotes.size();
        else
            return 0;
    }
    static class ViewHolder extends RecyclerView.ViewHolder{
        TextView textView_ttle, textView_desc;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            textView_ttle=itemView.findViewById(R.id.tv_vttle);
            textView_desc=itemView.findViewById(R.id.tv_vdesc);
        }
    }
}

File: app/src/main/java/com/example/todoapp/adapter/NotesAdapter.kt
package com.example.todoapp.adapter

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.todoapp.R
import com.example.todoapp.clicklisteners.ItemClickListener
import com.example.todoapp.model.Notes

class NotesAdapter(val list: List<Notes>, val itemClickListener: ItemClickListener) : RecyclerView.Adapter<NotesAdapter.ViewHolder>() {
    //private val listnotes: List<Notes>?
    //private val itemClickListener: ItemClickListener
    /*public void setListnotes(List<Notes> list){
       this.listnotes=list;
   }*/
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.notes_adapter_layout, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: NotesAdapter.ViewHolder, position: Int) { //we set data from list to layout here
        val notes = list[position]
        val ttle = notes.title
        val desc = notes.desc
        holder.textView_ttle.text = ttle
        Log.d("TAGVAL", "ttle$ttle")
        Log.d("TAGVAL", "ttle$desc")
        holder.textView_desc.text = desc
        holder.itemView.setOnClickListener(object : View.OnClickListener{
            override fun onClick(v: View?) {
                itemClickListener.OnClick(notes)
            }

        })
    }

    override fun getItemCount(): Int {
        return list.size ?:0
        //see elvs
    }

   inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var textView_ttle: TextView=itemView.findViewById(R.id.tv_vttle)
        var textView_desc: TextView=itemView.findViewById(R.id.tv_vdesc)


    }

}